### DISCUSSION TITLE
Cinnamon Menu Applet - Center Over Panel Setting

### DISCUSSION THREAD

**Discussion by amarraff** [linuxmint/discussions#555] [Ideas]
You can achieve a perfectly centered menu by moving the menu applet to the panel's center zone by itself. This only works properly if the panel is on the top or bottom of the display; if on the left or right sides, the top of the menu aligns with the applet instead of its center. Also, because the menu's position is based on the applet's position, centering breaks if you put your grouped window list next to the menu applet; the menu applet becomes pushed off-center, therefore the menu as well. 

Enabling the proposed centering setting would cause the menu to, when opened, be aligned with the center of the panel's position on the active display instead of the menu applet's position. This should completely ignore what zone the menu applet is in. I think the best way to present this in the menu applet's configure GUI would be a "center over panel" switch that is off by default. I think a good place for this setting would be under Configure (From right-click of menu applet) → Panel → Behavior → Center over panel. 

"Center to panel" may be a more grammatically correct title for the setting; the menu may not technically be "over" the panel, depending on the panel's location (if the panel is at the top of the display, the menu appears under the panel).

Setting off: Menu location varies based on applet position (current default behavior).
Setting on: Menu location is centered based on panel position.

![center_over_panel_OFF_panel_bottom 2024-07-30 11-40-26](https://github.com/user-attachments/assets/25e93967-8b94-43ec-aa37-b398b509fe5f)
![center_over_panel_ON_panel_bottom 2024-07-30 11-40-26](https://github.com/user-attachments/assets/b902a7d5-fe08-43da-845b-f5b0e7997395)
![center_over_panel_OFF_panel_left 2024-07-30 11-49-03](https://github.com/user-attachments/assets/251e6a3b-1fa3-4f51-b631-938d1510252c)
![center_over_panel_ON_panel_left 2024-07-30 11-49-03](https://github.com/user-attachments/assets/760fcbd2-2e42-4715-8b6c-9678220d64e9)

**Comment by Susensio** [comment]
I like it. It could be a floating option as well, centered in the screen

**Reply by amarraff** [reply]
> I like it. It could be a floating option as well, centered in the screen

That's a cool idea too! Could be implemented with a "center to display" option.

**Reply by anaximeno** [reply]
making it floating would probably be a theme change, by adding a margin between the menu and the panel.

**Comment by Leniwcowaty** [comment]
A bit necroposting, but digging this up again.

This would be a VERY welcome feature, allowing to create a Windows 11 - like look, with menu and pinned applications in the center of the panel, and the menu opening ALWAYS centered horizontally on the screen. Either automatically if it's placed in the center section of the panel, or manually with a switch mentioned in the issue.

I think this is worth looking at, since Cinnamon Menu is getting a rewrite anyways.

**Reply by Leniwcowaty** [reply]
Not letting this die and fade out - would be great to hear about that from Mint team BEFORE they finalize work on the new Menu applet

**Comment by legheed** [comment]
that the menu does not open centered really bugs  me out, took me some time to figure out that it is _supposed_ to be not in the middle of the screen but centered above the menu icon. sure, mouse movement wise it's economical, but it's hard to accept that the menu does not open at the same place every time, because it is dependent on how many symbols currently are on the task bar.

**Reply by amarraff** [reply]
This was exactly my problem, glad I'm not alone 🙂

**Reply by ost-guy** [reply]
Look like mx linux xfce

**Comment by LouVache** [comment]
There's an extension for that: https://cinnamon-spices.linuxmint.com/extensions/view/110
